availability/storage.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2016: Mirantis Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import json
+import logging
+
+import elasticsearch
+
+from availability import config
+
+LOG = logging.getLogger("storage")
+LOG.setLevel(config.get_config().get("logging", {}).get("level", "INFO"))
+
+
+def get_mapping(index, shards=2):
+    """Compose mapping for region services availability index."""
+    return {
+        "settings": {
+            "number_of_shards": shards
+        },
+        "mappings": {
+            index: {
+                "_all": {"enabled": False},
+                "properties": {
+                    "name": {"type": "keyword"},
+                    "region": {"type": "keyword"},
+                    "url": {"type": "text"},
+                    "time": {"type": "date"},
+                    "status": {"type": "boolean"}
+                }
+            }
+        }
+    }
+
+
+def check_create_elasticsearch_index(nodes, index):
+    """Create index if missed and return Elasticsearch instance.
+
+    :param nodes: list of elasticsearch nodes
+    :param index: index name
+    :returns: Elasticsearch or None on failure
+    :rtype: elasticsearch.Elasticsearch
+    :trype: None
+    """
+    try:
+        es = elasticsearch.Elasticsearch(nodes)
+        if not es.indices.exists(index):
+            mapping = json.dumps(get_mapping(index))
+            LOG.info("Creating Elasticsearch index: %s" % index)
+            es.indices.create(index=index, body=mapping)
+    except Exception as e:
```

Comment:
looks like we can just catch `elasticsearch.exceptions.ElasticsearchException` here instead of general exception.

Also, the only line that can throw smth is 62'd, so it would make sense to narrow the try/except to that line. Plus — you now have 2 functions that create & return an elasticsearch instance, a bit weird design decision. How about this function would just accept elasticsearch.Elasticsearch instead of nodes and re-use get_elasticsearch instead?